django/contrib/contenttypes/fields.py

```diff
@@ -236,21 +218,29 @@ def __get__(self, instance, cls=None):
         if instance is None:
             return self
 
+        # Make sure to use ContentType.objects.get_for_id() to ensure that
```

Comment:
I believe it's a reference to `ContentTypeManager`'s `get_for_*` methods cache. The actual `get_for_id()` calls happen in `self.get_content_type()` below.